# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Allocator.cpp
# Commit: d5ff86bdc1d4
# Full Hash: d5ff86bdc1d49356f9cf92eeb96009e9fa5045f5
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2024-10-22 21:31:58
# Description:
#   Bug 1925197 - Add missing nullptr check when allocating chunks r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D226121
# ==============================================================================

diff -r 225d63e189db -r d5ff86bdc1d4 js/src/gc/Allocator.cpp
--- a/js/src/gc/Allocator.cpp	Tue Oct 22 07:38:56 2024 +0000
+++ b/js/src/gc/Allocator.cpp	Tue Oct 22 08:24:58 2024 +0000
@@ -551,6 +551,10 @@
 
 ArenaChunk* GCRuntime::takeOrAllocChunk(AutoLockGCBgAlloc& lock) {
   ArenaChunk* chunk = getOrAllocChunk(lock);
+  if (!chunk) {
+    return nullptr;
+  }
+
   emptyChunks(lock).remove(chunk);
   return chunk;
 }
